summingbird-storm/src/main/scala/com/twitter/summingbird/storm/ComponentProvider.scala

```diff
@@ -0,0 +1,32 @@
+package com.twitter.summingbird.storm
+
+import com.twitter.algebird.Semigroup
+import com.twitter.summingbird.batch.Batcher
+import com.twitter.summingbird.online.executor.KeyValueShards
+import com.twitter.summingbird.storm.builder.Topology
+
+/**
+ * This trait describes how to build topology component which corresponds to `Producer`s chain.
+ */
+private[storm] trait ComponentProvider {
+  import StormTopologyBuilder._
+
+  /**
+   * Assuming this component corresponds to `Producer[Storm, T]` (which means it emits `Item[T]`)
```

Comment:
Any suggestions on how to describe this?

Basically this two methods correspond to non aggregated and aggregated use cases for `flatMap` and `spout`. I guess abstraction here is a little bit foggy unfortunately, but I didn't come up with better way to abstract our use cases.